tests/integration_tests/build/test_unittests.py

```diff
@@ -8,38 +8,45 @@
   is fixed
 """
 
-import os
-
-from subprocess import run
+import itertools
+import platform
 
 import pytest
 
-import host_tools.cargo_build as host  # pylint: disable=import-error
+import host_tools.cargo_build as host  # pylint:disable=import-error
 
+FEATURES = ["", "vsock"]
 
-CARGO_UNITTEST_REL_PATH = os.path.join(host.CARGO_BUILD_REL_PATH, "test")
+MACHINE = platform.machine()
+TARGETS = ["{}-unknown-linux-gnu".format(MACHINE),
+           "{}-unknown-linux-musl".format(MACHINE)]
 
 
-@pytest.mark.timeout(240)
-def test_unittests(test_session_root_path):
-    """Run unit and doc tests from all crates using default target."""
-    run(
-        'CARGO_TARGET_DIR={} RUST_TEST_THREADS=1 RUST_BACKTRACE=1 cargo test '
-        '--all --no-fail-fast'.format(
-            os.path.join(test_session_root_path, CARGO_UNITTEST_REL_PATH),
-        ),
-        shell=True,
-        check=True
-    )
+@pytest.mark.parametrize(
+    "features, target",
+    itertools.product(FEATURES, TARGETS)
+)
+def test_unittests(test_session_root_path, features, target):
```

Comment:
There is no need to have `features` as a cartesian product. This can be optimized as follows:
- paramterize only based on targets (musl vs gnu)
- in the test body check if the platform is x86_64 and if that's the case add an extra parameter to cargo test `--all-features`.

This would result in running the following tests:
- [x86_64] cargo test --all-features --target unknow-gnu -> we run the vsock tests as well
- [x86_64] cargo test --all-features --target unkown-musl -> we run the vsock tests as well
- [aarch_64] cargo test --target unknow-gnu -> only the default tests, no vsock
- [aarch_64] cargo test --target unknown-musl -> only the default tests, no vsock

Running `cargo test --features vsock` includes the tests being run only with `cargo test` so we are basically running all default tests twice.